src/lib.rs

```diff
@@ -106,6 +107,15 @@ pub fn libRustBCA(py: Python, m: &PyModule) -> PyResult<()> {
     Ok(())
 }
 
+pub fn find_index(Z: f64, Z2: &[f64]) -> Option<usize> {

```

Comment:
Actually, one more thought - indexing by Z could present a problem. What happens when someone wants to simulate a target of, for example, 90% Li-6 and 10% Li-7?